codeclimate-wrapper

```diff
@@ -18,32 +18,52 @@ EOF
   exit 1
 }
 
+docker_for_mac() {
+  test -S /var/tmp/docker.sock > /dev/null 2>&1
+}
+
+socket_exists() {
+  test -S $socket_path > /dev/null 2>&1
+}
+
+socket_path() {
```

Comment:
This approach still won't work for docker-machine (or boot2docker). The invocation at L47 will evaluate on the local machine and `exit 1` (which should be `return 1` anyway) since the socket is in the VM. That function is also incorrect anyway. This'll trip over into `socket_missing` when it shouldn't.

The invocations of `socket_exists` at L53 and L60 would also fail for similar reasons, but I don't think it'd get that far.

Something like this should work:

```sh
socket_path=/var/run/docker.sock

if [ -S /var/tmp/docker.sock ]; then
  socket_path=/var/tmp/docker.sock
fi
```

Leaving the rest of the script completely as-is, except for replacing the hardcoded socket with `$socket_path`.

This is very intentional in _not_ moving things out into named functions like `socket_path` or `socket_exists` because that'd change the semantics: currently, we're giving arguments to the `ssh` commands, your changes make them actual local invocations _to produce_ arguments for the `ssh` commands -- which won't work.